tests/components/azure_data_explorer/test_init.py

```diff
@@ -0,0 +1,355 @@
+"""Test the init functions for AEH."""
+from datetime import timedelta
+import logging
+from unittest.mock import patch
+
+from azure.kusto.data.exceptions import KustoAuthenticationError, KustoServiceError
+import pytest
+
+from homeassistant.components import azure_data_explorer
+from homeassistant.components.azure_data_explorer.const import (
+    CONF_SEND_INTERVAL,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from .conftest import FilterTest
+from .const import (
+    AZURE_DATA_EXPLORER_PATH,
+    BASE_CONFIG_FULL,
+    BASIC_OPTIONS,
+    IMPORT_CONFIG,
+)
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def test_put_event_on_queue_with_managed_client(
+    hass,
+    entry_managed,
+    mock_azure_data_explorer_ManagedStreamingIngestClient_ingest_data,
+) -> None:
+    """Test listening to events from Hass. and writing to ADX with managed client."""
+
+    hass.states.async_set("sensor.test_sensor", STATE_ON)
+
+    async_fire_time_changed(
+        hass, utcnow() + timedelta(seconds=entry_managed.options[CONF_SEND_INTERVAL])
+    )
+
+    await hass.async_block_till_done()
+    mock_azure_data_explorer_ManagedStreamingIngestClient_ingest_data.assert_called_once()
```

Comment:
These lines are duplicated in the tests below. Can we make a common function or a parameterized test out of them